pkg/istio-agent/sds-agent.go

```diff
@@ -251,22 +153,21 @@ type AgentConfig struct {
 //
 // If node agent and JWT are mounted: it indicates user injected a config using hostPath, and will be used.
 //
-func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig) *Agent {
-	a := &Agent{
+func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig, sopts *security.Options) *Agent {
+	sa := &Agent{
 		proxyConfig: proxyConfig,
 		cfg:         cfg,
+		secOpts:     sopts,
 	}
 
-	discAddr := proxyConfig.DiscoveryAddress
+	// Fix the defaults - mainly for tests ( main uses env )
+	if sopts.RecycleInterval.Seconds() == 0 {
+		sopts.RecycleInterval = 5 * time.Minute
```

Comment:
can remove this hacky way

Reply:
Well, it's not really hacky - quite common to set defaults if the values are not set. 
This will go away when we move to the XDS impl.